File: python/003.py
# Largest Prime Factor

import math

def is_prime(n):
    if n <= 1:
        return False
    
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    
    return True

x, f = 600851475143, 2

while x > 1:
    if not is_prime(f):
        f += 1
        continue

    if x % f == 0:
        x //= f
    else:
        f += 1

print(f)

File: rust/003.rs
// Largest Prime Factor

fn main() {
    let mut x = 600851475143u64;
    let mut i = 2u64;

    loop {
        if x <= 1 {
            break;
        }

        if x % i == 0 {
            x /= i;
        } else {
            i += 1;
        }
    }

    println!("{}", i);
}
